fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -179,10 +182,15 @@ public OptExpression rewrite() {
         ScalarOperator mvPredicate = MvUtils.rewriteOptExprCompoundPredicate(mvExpression, mvColumnRefRewriter);
 
         if (materializationContext.getMvPartialPartitionPredicate() != null) {
-            // add latest partition predicate to mv predicate
-            ScalarOperator rewritten =
-                    mvColumnRefRewriter.rewrite(materializationContext.getMvPartialPartitionPredicate());
-            mvPredicate = MvUtils.canonizePredicate(Utils.compoundAnd(mvPredicate, rewritten));
+            List<ScalarOperator> partitionPredicates =
+                    getPartitionRelatedPredicates(mvPredicate, mvRewriteContext.getMaterializationContext().getMv());
+            if (partitionPredicates.stream().noneMatch(p -> p instanceof IsNullPredicateOperator)) {
```

Comment:
If partition column has predicate `isNull`, mv will lost partition predicate even if some partitions are outdated?

Reply:
no, if it has isNull predicate, the partition with null partition is updated, then the mv can not be used to rewrite. or the partial predicates are useless. so here do not add the partial predicates.